expressjs/src/routes/events/printer.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @typedef {import('cloudevents').CloudEvent} CloudEvent
+ */
+
+class PrettyPrinter {
+
+  /**
+   * Prints a CloudEvent in human-readable format.
+   *
+   * @param {CloudEvent} ce - the CloudEvent to receive
+   * @returns {string} - the human-readable representation of the CloudEvent
+   */
+  print(ce) {
+    let valid = 'valid'
+    try {
+      ce.validate()
+    } catch (err) {
+      valid = 'invalid'
+    }
+    let buf = '☁️  cloudevents.Event\n'
+    buf += `Validation: ${valid}\n`
+    const attr = attributes(ce)
+    buf += printAttr(attr)
+    buf += printExts(ce, attr)
+    buf += printData(ce)
+    return buf
+  }
+
+}
+
+/**
+ * @param {CloudEvent} ce - the CloudEvent
+ * @returns {Object} - the CloudEvent attributes
+ * @private
+ */
+function attributes(ce) {
+  const attr = {
+    specversion: ce.specversion,
+    type: ce.type,
+    source: ce.source,
+    id: ce.id,
+  }
+  if (ce.subject) {
+    attr.subject = ce.subject
+  }
+  if (ce.time) {
+    attr.time = ce.time
+  }
+  if (ce.datacontentencoding) {
+    attr.datacontentencoding = ce.datacontentencoding
+  }
+  if (ce.dataschema) {
+    attr.dataschema = ce.dataschema
+  }
+  if (ce.datacontenttype) {
+    attr.datacontenttype = ce.datacontenttype
+  }
+  return attr
+}
+
+/**
+ * @param {Object} attr - the CloudEvent attributes
+ * @returns {string} - the buffer
+ */
+function printAttr(attr) {
+  let buf = 'Context Attributes,\n'
+  for (const [k, v] of Object.entries(attr)) {
+    buf += `  ${k}: ${v}\n`
+  }
+  return buf
+}
+
+/**
+ * @param {CloudEvent} ce - the CloudEvent
+ * @param {Object} attr - the CloudEvent attributes
+ * @returns {Object} - the CloudEvent extensions
+ * @private
+ */
+function extensions(ce, attr) {
+  const excludes = Object.keys(attr)
+  excludes.push('data')
+  excludes.push('data_base64')
+  const exts = {}
+
+  for (const [k, v] of Object.entries(ce)) {
+    if (!excludes.includes(k) && v) {
+      exts[k] = v
+    }
+  }
+  return exts
+}
+
+/**
+ * @param {CloudEvent} ce - the CloudEvent
+ * @param {Object} attr - the CloudEvent attributes
+ * @returns {string} - the buffer
+ */
+function printExts(ce, attr) {
+  let buf = ''
+  const exts = extensions(ce, attr)
+  if (Object.keys(exts).length > 0) {
+    buf += `Extensions,\n`
```

Comment:
Should there be `:` after Extensions?
See my previous comment about the output.